### ISSUE TITLE
500 Internal server error building clientLibs from cloudtools endpoints framework maven

### ISSUE THREAD

**Issue by JolietJake** [GoogleCloudPlatform/endpoints-framework-gradle-plugin#106]
I'm building endpoints for an appengine standard server with google cloud tools.  I'm able to generate openapidocs, Discoverydocs, but when I try and generate clientLibs with --- endpoints-framework-maven-plugin:2.0.1:clientLibs (default-cli) @ mobile ---
[INFO] Endpoints Tool params : [get-client-lib, -o, C:\Users\steve.......:

I get:

ints Tool Error: 500 Internal Server Error
[ERROR] {"error": {"code": 500, "message": "Internal Server Error", "errors": [{"debug_info": "Neither servicePath nor rpcPath is defined.", "message": "unknown exception"}]}}
[ERROR] -> [Help 1]

this problem looks familiar to https://github.com/GoogleCloudPlatform/endpoints-framework-gradle-plugin/issues/98.  That was solved at the Google end.

Any help appreciated.

**Comment by elharo** [comment]
@tangiel @chingor13 did the config change again?

**Comment by chingor13** [comment]
Nothing has changed on the server side for quite some time. This is a separate error. Can you see if the discovery doc being pushed to the library generation endpoint has the `servicePath` or `rpcPath` fields defined?

**Comment by JolietJake** [comment]
I checked the endpoints

 

"servicePath": "webholesendpoint/v2/",

"version": "v2"

}

 

All the ones I added were fine.  However, the default skeleton one which the template created – MyFirstAPI.java, which I had left in there, did not have a servicePath.  It was <null>, no default apparently.  I deleted it from the java directory and removed it from the web.xml.

 

The clientLibs then built!  Thank you for the assistance.
